x/ibc/04-channel/keeper/handshake.go

```diff
@@ -169,11 +180,13 @@ func (k Keeper) ChanOpenAck(
 		)
 	}
 
-	// TODO: blocked by #5542
-	// key := sdk.NewKVStoreKey(portID)
-	// if !k.portKeeper.Authenticate(key, portID) {
-	// 	return sdkerrors.Wrap(port.ErrInvalidPort, portID)
-	// }
+	if _, ok := k.GetChannelCapability(ctx, portID, channelID); !ok {
+		return sdkerrors.Wrap(types.ErrChannelCapabilityNotFound, "channel capability not found")
+	}
+
+	if !k.portKeeper.Authenticate(ctx, portCap, portID) {
```

Comment:
This can be removed, the channel capability is used for `ChanOpenAck` and `ChanOpenConfirm`